src/universal/modules/invitation/containers/Invitation/Invitation.js

```diff
@@ -0,0 +1,154 @@
+        // If user already has an account, let them accept the new team via the UI:
+        dispatch(push('/me'));
+      } else if (user && user.profile.isNew === true) {
+        // If the user is new let's process their invite:
+        this.processInvitation();
+      }
+    }
+  }
+
+  processInvitation = () => {
+    const {dispatch, inviteToken} = this.props;
+    const options = {
+      variables: {
+        inviteToken
+      }
+    };
+    cashay.mutate('acceptInvitation', options).then(({data, error}) => {
+      if (error) {
+        if (error.subtype === 'alreadyJoined') {
+          /*
+           * This should be *very* difficult to have occur:
+           */
+          dispatch(showError({
+            title: 'Team already joined',
+            message: `
+              Hey, we think you already belong to this team.
+            `,
+            action: {
+              label: 'Ok',
+            },
+            autoDismiss: 0
+          }));
+          dispatch(push('/settings/me'));
+          return;
+        } else if (error.subtype === 'invalidEmail') {
+          const {user: {email}} = this.props;
+          dispatch(showError({
+            title: 'Invitation invalid',
+            message: `
+              That invitation isn't valid for the email address
+              ${email}. Perhaps switch accounts or ask for a new invitation?
+            `,
+            action: {
+              label: 'Ok',
+            },
+            autoDismiss: 0
+          }));
+        } else if (error.subtype === 'invalidToken') {
+          dispatch(showError({
+            title: 'Invitation invalid',
+            message: `
+              We had difficulty with that link. Did you paste it correctly?
+            `,
+            action: {
+              label: 'Ok',
+            },
+            autoDismiss: 10
+          }));
+        } else if (error.subtype === 'notFound') {
+          dispatch(showWarning({
+            title: 'Invitation not found, but don\'t worry',
+            message: `
+              Hey we couldn't find that invitation. If you'd like to
+              create your own team, you can start that process here.
+            `,
+            action: {
+              label: 'Got it',
+            },
+            autoDismiss: 0
+          }));
+        } else {
+          console.warn('unable to accept invitation:');
+          console.warn(error);
+        }
+        // TODO: pop them a toast and tell them what happened?
+        dispatch(push('/welcome'));
+      } else if (data) {
+        const {id} = data.acceptInvitation.team;
+        dispatch(setWelcomeActivity(`/team/${id}`));
+        dispatch(showSuccess({
+          title: 'You\'re in!',
+          message: `
+            Welcome to Action. Let's get you set up.
+          `
+        }));
+        dispatch(push('/me/settings'));
+      }
+    }).catch(console.warn.bind(console));
+  };
+
+  renderLogin = () => (
+    <div>
```

Comment:
This appears under the auth0 lock. This is where I've asked @ackernaut for a little design help.